### PR TITLE
fix: wrapping set focus in dom.queueupdate

### LINKED ISSUE #3685: issue with autofocus in text-field
**Describe the bug; what happened?**

When I use a `<fast-text-field>` with `autofocus` (`<fast-text-field autofocus>`) I get issues with the rest of the application. My use case is an Aurelia 2 project where using the `autofocus` interferes with the routing. I had a similar issue when using `autofocus` inside a `<fast-dialog>` where the autofocus would not work.

In both scenario, if I do the focus manually myself after a 50ms timeout (with `myElement.focus()`) then it works as it should.

[I suspect that the `focus()` is called a little too early here. ](https://github.com/microsoft/fast/blob/2879571796f210da49be8df90a74faee429bd34b/packages/web-components/fast-foundation/src/text-field/text-field.ts#L208)

**What are the steps to reproduce the issue?**

* Maybe set up a page with a `<fast-text-field autofocus>` inside a `<fast-dialog hidden>`
* And then show the fast dialog and check if everything works fine, including the focus of the text field

**What behavior did you expect?**

* autofocus should work
* autofocus should not cause any issues with the rest of the app

**In what environment did you see the issue?**

* MacOS
* Chrome
* Aurelia 2 project

### PR DESCRIPTION
**Pull request by marjonlynch**
## Description
Autofocus seems to work but in the context of other wrapping frameworks it might be happening too soon.
potentially closes #3685 

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [x] **Bug fix**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

**Adding or modifying component(s) in `@microsoft/fast-components` checklist**

- [ ] I have added a new component
- [x] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist

- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

File: packages/web-components/fast-foundation/src/text-field/text-field.ts [hunk 1]
```diff
@@ -204,9 +204,11 @@ export class TextField extends FormAssociated<HTMLInputElement> {
 
         this.proxy.setAttribute("type", this.type);
 
-        if (this.autofocus) {
-            this.focus();
-        }
+        DOM.queueUpdate(() => {
```
Comment (scomea) [inline packages/web-components/fast-foundation/src/text-field/text-field.ts:207]: Can you check autofocus before queuing an update?  I don't think we want all text fields queing something.

Comment (chrisdholt) [reply packages/web-components/fast-foundation/src/text-field/text-field.ts:207]: +1 - would suggest we only do this in the case autofocus is true.

Comment (marjonlynch) [reply packages/web-components/fast-foundation/src/text-field/text-field.ts:207]: this makes sense, will change

Comment (marjonlynch) [reply packages/web-components/fast-foundation/src/text-field/text-field.ts]: I'm assuming nothing can change with regard to this.autofocus between checking and queuing the update, so only 1 check before the queue call

**Pull request merged**